benchmarktests/target_auth_gcp.go

```diff
@@ -0,0 +1,294 @@
+	_, err = client.Logical().Write("auth/"+authPath+"/config", GCPAuthConfig)
+
+	if err != nil {
+		return nil, fmt.Errorf("error writing gcp config: %v", err)
+	}
+
+	setupLogger.Trace(parsingConfigLogMessage("role"))
+	GCPRoleConfig, err := structToMap(config.GCPTestRoleConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing role config from struct: %v", err)
+	}
+
+	// Write GCP Role
+	setupLogger.Trace(writingLogMessage("role"), "name", config.GCPTestRoleConfig.Name)
+	_, err = client.Logical().Write("auth/"+authPath+"/role/"+config.GCPTestRoleConfig.Name, GCPRoleConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error writing gcp role: %v", err)
+	}
+
+	// Select one of the configured service accounts if more than 1
+	rand.Seed(time.Now().Unix())
+
+	var serviceAccount string
+
+	if len(config.GCPTestRoleConfig.BoundServiceAccounts) > 0 {
+		n := rand.Int() % len(config.GCPTestRoleConfig.BoundServiceAccounts)
+		serviceAccount = config.GCPTestRoleConfig.BoundServiceAccounts[n]
+	}
+
+	iam := config.GCPTestRoleConfig.Type == "iam"
+	jwt, err := getSignedJwt(g.config.Config.GCPTestRoleConfig.Name, config.GCPAuthConfig.Credentials, config.GCPTestRoleConfig.MaxJWTExp, serviceAccount, iam)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching JWT: %v", err)
+	}
+
+	return &GCPAuth{
+		header:     generateHeader(client),
+		pathPrefix: "/v1/" + filepath.Join("auth", authPath),
+		jwt:        jwt,
+		roleName:   g.config.Config.GCPTestRoleConfig.Name,
+		timeout:    g.timeout,
+		logger:     g.logger,
+	}, nil
+}
+
+// Func Flags accepts a flag set to assign additional flags defined in the function
+func (g *GCPAuth) Flags(fs *flag.FlagSet) {}
+
+func getSignedJwt(role string, jwtCreds string, jwtExp string, serviceAccount string, iam bool) (string, error) {
+	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, cleanhttp.DefaultClient())
+
+	credentials, tokenSource, err := gcputil.FindCredentials(jwtCreds, ctx, iamcredentials.CloudPlatformScope)
+	if err != nil {
+		return "", fmt.Errorf("could not obtain credentials: %v", err)
+	}
+
+	httpClient := oauth2.NewClient(ctx, tokenSource)
+
+	if serviceAccount == "" && credentials != nil {
+		serviceAccount = credentials.ClientEmail
+	}
+
+	if !iam {
+		// Check if the metadata server is available.
+		if !metadata.OnGCE() {
+			return "", fmt.Errorf("could not obtain service account from credentials (are you using Application Default Credentials?). You must provide a service account to authenticate as")
+		}
+		metadataClient := metadata.NewClient(cleanhttp.DefaultClient())
+		v := url.Values{}
+		v.Set("audience", fmt.Sprintf("http://vault/%s", role))
+		v.Set("format", "full")
+		path := "instance/service-accounts/default/identity?" + v.Encode()
+		instanceJwt, err := metadataClient.Get(path)
+		if err != nil {
+			return "", fmt.Errorf("unable to read the identity token: %w", err)
+		}
+		return instanceJwt, nil
+
+	} else {
+		ttl := time.Duration(15) * time.Minute
```

Comment:
Won't our credentials become invalid if the bench runs longer?

Reply:
yea I see your point. I was assuming that if they set the overall duration to something large, like 20m, then they would have to adjust their gcp role configuration to include `max_jwt_exp: "21m"`. Otherwise I think we'd need to alter this setup function to pass in the duration: https://github.com/hashicorp/vault-benchmark/blob/main/benchmarktests/benchmark_targets.go#L33

Reply:
Do we have access to the configured duration of the test? If so we could pass in that value here.